### PR TITLE
Allow to use Regexps to permanent redirect paths

### PR DESCRIPTION
**Pull request by spickermann**
@timoschilling I would like to use regexp pattern to redirect routes.

### CODE REVIEW

**Review by timoschilling — APPROVED** [review]

File: lib/permanent_not_found.rb [hunk 1]
```diff
@@ -4,23 +4,28 @@
...
     else
       @app.call(env)
     end
   end
 
   private
 
-  def response
+  def permanent_redirected?(path)
+    @paths.include?(path) || @pattern =~ path
```
Comment (JanBe) [inline lib/permanent_not_found.rb]: Should this maybe be split into 2 further methods in order to clarify, that 2 different methods for verifying the inclusion of the path in the filter are used?

Comment (lagr) [inline lib/permanent_not_found.rb]: Nah. I find it quite easy to grasp from this line.

Comment (JanBe) [inline lib/permanent_not_found.rb]: Just came back here to say this, as I thought about it again while cooking xD
Two methods would probably just make it overly complex

Comment (timoschilling) [inline lib/permanent_not_found.rb]: Maybe we should not use a method at all to improve the call speed ;)

Comment (spickermann) [inline lib/permanent_not_found.rb]: 1bae09fb2ecdd32e63311526238fb66456ffd699a857396004d37ccf0098303a

**Pull request merged**